test/cases/literals/heredoc-dollar-no-lead-space.hack

```diff
@@ -0,0 +1,4 @@
+// This test currently fails but should pass
```

Comment:
If we add the same is-identifier-char check that you added to scanner.cc t[he first time we look  for `$` in `scan_body`](https://github.com/antoniodejesusochoasolano/tree-sitter-hack/blob/81944f086839fd70d27d0396c75e080eb516490c/src/scanner.cc#L214-L217), I think that fixes this issue. Something like:

```c
if (!is_nowdoc && prefix.empty()) {
  if (peek() == '{') next();
  if (peek() == '$') {
    next();
    if (is_identifier_char(peek())) {
      return false;
    }
  }
}
```